src/families/hedera/utils.ts

```diff
@@ -11,23 +17,28 @@ export async function calculateAmount({
   transaction: Transaction;
 }): Promise<{
   amount: BigNumber;
-  estimatedFees: BigNumber;
   totalSpent: BigNumber;
 }> {
-  // NOTE: Hedera declares stable fees in USD
-  //       If we can get the current USD/HBAR price here..
-  //       > transfer fee is 0.0001 USD
-  const estimatedFees = new BigNumber("83300"); // 0.000833 ℏ (as of 2021-09-20)
+  let amount;
+
+  if (transaction.useAllAmount === true) {
+    amount = await estimateMaxSpendable({ account });
+  } else {
+    const txAmount = transaction.amount.plus(estimatedFees);
 
-  const amount =
-    transaction.useAllAmount == true
-      ? await estimateMaxSpendable({ account })
-      : transaction.amount;
+    if (txAmount.isGreaterThan(account.balance)) {
+      // NOTE: work-around for 'NotEnoughBalance' edge case
+      // since the addition of estimated fees overflows account balance,
+      // set amount w/o the added estimated fees.
+      amount = transaction.amount;
+    } else {
+      amount = transaction.amount.plus(estimatedFees);
```

Comment:
This looks really weird, like you're ignoring the fees just because you can't pay them. :thinking: seems it could lead to other issues e.g. signing an amount different from what the user set for the transaction.

Let's merge and see what the bot has to say about it. :slightly_smiling_face: